torch/linalg/__init__.py

```diff
@@ -658,60 +658,69 @@
-Vh`. If :attr:`input` is a batch of tensors, then ``U``, ``S``, and ``Vh`` are
+namedtuple ``(U, S, Vh)``, such that
+:math:`\texttt{input} = \texttt{U} \operatorname{diag}(\texttt{S}) \texttt{Vh}`.
+If :attr:`input` is a batch of matrices, then ``U``, ``S``, and ``Vh`` are
 also batched with the same batch dimensions as :attr:`input`.
 
-If :attr:`full_matrices` is ``False`` (default), the method returns the reduced singular
-value decomposition i.e., if the last two dimensions of :attr:`input` are
-``m`` and ``n``, then the returned `U` and `V` matrices will contain only
-:math:`min(n, m)` orthonormal columns.
+If :attr:`full_matrices` is ``False``, the method returns the reduced singular
+value decomposition. In this case, if the last two dimensions of :attr:`input` are
+``m`` and ``n``, then the returned ``U`` and ``Vh`` matrices will contain only
+``min(n, m)`` orthonormal columns.
 
-If :attr:`compute_uv` is ``False``, the returned `U` and `Vh` will be empy
+If :attr:`compute_uv` is ``False``, the returned ``U`` and ``Vh`` will be empty
 tensors with no elements and the same device as :attr:`input`. The
 :attr:`full_matrices` argument has no effect when :attr:`compute_uv` is False.
 
-The dtypes of ``U`` and ``V`` are the same as :attr:`input`'s. ``S`` will
+The dtypes of ``U`` and ``Vh`` are the same as :attr:`input`'s. ``S`` will
 always be real-valued, even if :attr:`input` is complex.
 
 .. note:: Unlike NumPy's ``linalg.svd``, this always returns a namedtuple of
-          three tensors, even when :attr:`compute_uv=False`. This behavior may
+          three tensors, even when :attr:`compute_uv` is ``False``. This behavior may
           change in a future PyTorch release.
 
 .. note:: The singular values are returned in descending order. If :attr:`input` is a batch of matrices,
-          then the singular values of each matrix in the batch is returned in descending order.
+          then the singular values of each matrix in the batch are returned in descending order.
 
-.. note:: The implementation of SVD on CPU uses the LAPACK routine `?gesdd` (a divide-and-conquer
-          algorithm) instead of `?gesvd` for speed. Analogously, the SVD on GPU uses the cuSOLVER routines
-          `gesvdj` and `gesvdjBatched` on CUDA 10.1.243 and later, and uses the MAGMA routine `gesdd`
-          on earlier versions of CUDA.
+.. note:: The implementation of SVD on CPU uses the LAPACK routine ``?gesdd`` (a divide-and-conquer
+          algorithm) instead of ``?gesvd`` for speed. Analogously, the SVD on GPU uses the
+          cuSOLVER routines ``gesvdj`` and ``gesvdjBatched`` on CUDA 10.1.243 and later,
+          and uses the MAGMA routine ``gesdd`` on earlier versions of CUDA.
 
-.. note:: The returned matrix `U` will be transposed, i.e. with strides
-          :code:`U.contiguous().transpose(-2, -1).stride()`.
+.. note:: The returned matrix ``U`` will not be contiguous. It will be represented as a
```

Comment:
```
.. note:: Each \`\`U\`\` matrix will be transposed (equivalently "column-major" or "Fortran-contiguous").
```

U may not be just a matrix, and it's odd that this note started by explaining what U was not instead of what it was. Saying the matrix or matrices are transposed may be more familiar to PyTorch users.

Reply:
The problem with this is that if you say that "U is transposed", some people might do `U, _, V = torch.svd(X); U = U.T` to "account for that".

I think that the people that will care about the strides of `U` will already know what strides are and what a contiguous tensor is. Everyone else will just omit this note.

About `U` not being a matrix, fair enough. I just used the word "matrix" because the previous note used it. I can change that to "The returned `U` will not be contiguous. The matrix (or batch of matrices) will be represented as column-major matrices (i.e. Fortran-contiguous)." What do you think?

Reply:
I appreciate your thinking and your suggestion sounds good.